diff --git a/Source/WTF/wtf/PlatformEnable.h b/Source/WTF/wtf/PlatformEnable.h
--- a/Source/WTF/wtf/PlatformEnable.h
+++ b/Source/WTF/wtf/PlatformEnable.h
@@ -629,7 +629,7 @@
 #define ENABLE_WEBGPU PLATFORM(COCOA)
 #endif
 
-#if !defined(ENABLE_WEBGPU_BY_DEFAULT) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 260000) || (PLATFORM(IOS)) || (PLATFORM(VISION)))
+#if !defined(ENABLE_WEBGPU_BY_DEFAULT) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 260000) || PLATFORM(IOS) || PLATFORM(VISION) || (PLATFORM(WIN) && ENABLE_WEBGPU))
 #define ENABLE_WEBGPU_BY_DEFAULT 1
 #endif
 
